src/vendor/github.com/xelabs/go-mysqlstack/sqlparser/sql.y

```diff
@@ -135,15 +135,29 @@ func forceEOF(yylex interface{}) {
 	FOR
 
 
+%type   <bytes>
+	index_type
+	index_type_opt
+	index_type_name
+	spatial_or_fulltext
+	constraint_keyword_opt
+
+%type   <str>
+	index_name
+
 // INDEX.
 %token	<bytes>
 	ALGORITHM
 	BTREE
+	HASH
```

Comment:
order

Reply:
Good advice, I changed it by alpha order. Please make a check.